tests/components/lektrico/test_config_flow.py

```diff
@@ -0,0 +1,168 @@
+    assert result.get("data") == {
+        CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        ATTR_SERIAL_NUMBER: MOCKED_DEVICE_SERIAL_NUMBER,
+        CONF_TYPE: MOCKED_DEVICE_TYPE,
+        ATTR_HW_VERSION: MOCKED_DEVICE_BOARD_REV,
+    }
+    assert "result" in result
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    assert result.get("result").unique_id == MOCKED_DEVICE_SERIAL_NUMBER
+
+
+async def test_user_setup_already_exists(
+    hass: HomeAssistant, mock_device_config
+) -> None:
+    """Test manually setting up when the device already exists."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+        unique_id=MOCKED_DEVICE_SERIAL_NUMBER,
+    )
+    entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_user_setup_device_offline(
+    hass: HomeAssistant, mock_device_config
+) -> None:
+    """Test manually setting up when device is offline."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    mock_device_config.side_effect = DeviceConnectionError
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {CONF_HOST: "cannot_connect"}
+
+
+# async def test_discovered_zeroconf(
+#     hass: HomeAssistant, mock_device_config, mock_setup_entry
+# ) -> None:
+#     """Test we can setup when discovered from zeroconf."""
+
+#     result = await hass.config_entries.flow.async_init(
+#         DOMAIN,
+#         context={"source": SOURCE_ZEROCONF},
+#         data=MOCKED_DEVICE_ZEROCONF_DATA,
+#     )
+#     assert result["type"] is FlowResultType.FORM
+#     assert result["errors"] is None
+
+#     result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+#     assert result2["type"] is FlowResultType.CREATE_ENTRY
+#     assert result2["data"] == {
+#         CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+#         ATTR_SERIAL_NUMBER: MOCKED_DEVICE_SERIAL_NUMBER,
+#         CONF_TYPE: MOCKED_DEVICE_TYPE,
+#         ATTR_HW_VERSION: MOCKED_DEVICE_BOARD_REV,
```

Comment:
Why is this commented out?

Reply:
I fixed it and it is now available. Thank you.